ref-resolver.js

```diff
@@ -11,8 +11,10 @@ class RefResolver {
     if (schema.$id !== undefined && schema.$id.charAt(0) !== '#') {
       schemaId = schema.$id
     }
-    this.insertSchemaBySchemaId(schema, schemaId)
-    this.insertSchemaSubschemas(schema, schemaId)
+    if (this.getSchema(schemaId) === undefined) {
```

Comment:
Is this changes itself self contained?
Do we need a test case for it?

Reply:
I didn't add this check. I just moved it inside the class.
https://github.com/fastify/fast-json-stringify/blob/b0bb1a699e5faed3657ad8c1d665920e29304be5/index.js#L113